R/object_overwrite_linter.R

```diff
@@ -76,7 +76,7 @@ object_overwrite_linter <- function(
   # test that the symbol doesn't match an argument name in the function
   # NB: data.table := has parse token LEFT_ASSIGN as well
   xpath_assignments <- glue("
-    //SYMBOL[
+    (//SYMBOL | //STR_CONST)[
```

Comment:
I wonder about the performance of `//STR_CONST` vs. `//LEFT_ASSIGN`. Actually per the original issue about avoiding `//expr`, they appear at roughly the same frequency, so probably it's not much difference.

https://github.com/r-lib/lintr/issues/1358#issue-1260855389

Reply:
Yeah, from what I remember, only `//expr` stood out to be exceptionally bad cost-wise.